py/builtinimport.c

```diff
@@ -438,7 +439,14 @@ STATIC mp_obj_t process_import_at_level(qstr full_mod_name, qstr level_mod_name,
         }
         #endif // MICROPY_MODULE_OVERRIDE_MAIN_IMPORT
 
-        if (stat == MP_IMPORT_STAT_DIR) {
+        nlr_buf_t nlr;
+        if (nlr_push(&nlr)) {
```

Comment:
This doesn't look right. `nlr_push` is the equivalent of `try` in Python so would need to be before the code that can raise the exception.

Reply:
This sounds like a code style issue.  I'll do whatever is needed to get this merged, but I chose this arrangement so that I wouldn't have to increase the indentation of the ~27 lines of code we are catching exceptions from, making it harder to fit code and comments there.

Reply:
This is not a style issue, but rather a functional one. The code as it is currently will remove all modules from `sys.modules`, not just ones that fail to import.

Furthermore, the error handling should be in the `else` statement of the `if (nlr_push...` and the `nlr_pop` should be at the end of the `if` block (right before `} else {`).